src/NServiceBus.Msmq.AcceptanceTests/When_using_instance_mapping_file_with_conventions.cs

```diff
@@ -0,0 +1,129 @@
+                .Done(c => c.MessagesForInstance1 + c.MessagesForInstance2 >= 5)
+                .Run();
+
+            Assert.That(context.MessagesForInstance1, Is.EqualTo(0));
+            Assert.That(context.MessagesForInstance2, Is.EqualTo(5));
+        }
+
+        static string mappingFilePath = Path.Combine(TestContext.CurrentContext.TestDirectory, nameof(When_using_instance_mapping_file) + ".xml");
+        static string destination;
+
+        public class Context : ScenarioContext
+        {
+            public int MessagesForInstance1;
+            public int MessagesForInstance2;
+        }
+
+        public class SenderWithMappingFile : EndpointConfigurationBuilder
+        {
+            public SenderWithMappingFile()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var conventions = c.Conventions();
+                    conventions.DefiningCommandsAs(type => type.Namespace != null && type.Namespace.EndsWith("Commands"));
+
+                    var routingSettings = c.UseTransport<MsmqTransport>().Routing();
+                    routingSettings.RouteToEndpoint(typeof(Message), destination);
+                    routingSettings.InstanceMappingFile().FilePath(mappingFilePath);
+                });
+            }
+        }
+
+        public class ScaledOutReceiver : EndpointConfigurationBuilder
+        {
+            public ScaledOutReceiver()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    var conventions = c.Conventions();
+                    conventions.DefiningCommandsAs(type => type.Namespace != null && type.Namespace.EndsWith("Commands"));
+                });
+            }
+
+            public class MessageHandler : IHandleMessages<Message>
+            {
+                public MessageHandler(Context testContext, ReadOnlySettings settings)
+                {
+                    this.testContext = testContext;
+                    this.settings = settings;
+                }
+
+                public Task Handle(Message message, IMessageHandlerContext context)
+                {
+                    var instanceDiscriminator = settings.Get<string>("EndpointInstanceDiscriminator");
+                    if (instanceDiscriminator == "1")
+                    {
+                        Interlocked.Increment(ref testContext.MessagesForInstance1);
+                    }
+                    if (instanceDiscriminator == "2")
+                    {
+                        Interlocked.Increment(ref testContext.MessagesForInstance2);
+                    }
+
+                    return Task.FromResult(0);
+                }
+
+                Context testContext;
+                ReadOnlySettings settings;
+            }
+        }
+    }
+}
+
+namespace NServiceBus.Transport.Msmq.AcceptanceTests.Commands
+{
+    public class Message
```

Comment:
Does scanning pick up this type?

I added a test recently where I would have to tell the ATT Fx to pick it up explicitly since its outside of the test class?

https://github.com/Particular/NServiceBus/commit/b714915a87b79e2abfe82f45affcc3fab3453d06#diff-5e0eea387d62aef54e65aa3ade16738bR39

Or perhaps v6 behaves differently?